src/Raven.Server/Web/Studio/StudioCollectionsHandlerProcessorForPreviewRevisions.cs

```diff
@@ -0,0 +1,111 @@
+using JetBrains.Annotations;
+using Raven.Client;
+using Raven.Server.Documents;
+using Raven.Server.ServerWide.Context;
+using Raven.Server.Web.Studio.Processors;
+using Sparrow.Json;
+
+namespace Raven.Server.Web.Studio;
+
+internal sealed class StudioCollectionsHandlerProcessorForPreviewRevisions : AbstractStudioCollectionsHandlerProcessorForPreviewRevisions<DatabaseRequestHandler, DocumentsOperationContext>
+{
+    private int _start;
+    private int _pageSize;
+    private long _totalResults;
+
+    public StudioCollectionsHandlerProcessorForPreviewRevisions([NotNull] DatabaseRequestHandler requestHandler) : base(requestHandler)
+    {
+    }
+
+    protected override async Task InitializeAsync(DocumentsOperationContext context, CancellationToken token)
+    {
+        await base.InitializeAsync(context, token);
+
+        _start = RequestHandler.GetStart();
+        _pageSize = RequestHandler.GetPageSize();
+
+        _totalResults = string.IsNullOrEmpty(Collection)
+            ? RequestHandler.Database.DocumentsStorage.RevisionsStorage.GetNumberOfRevisionDocuments(context)
+            : RequestHandler.Database.DocumentsStorage.RevisionsStorage.GetNumberOfRevisionDocumentsForCollection(context, Collection);
+    }
+
+    protected override IDisposable OpenReadTransaction(DocumentsOperationContext context)
+    {
+        return context.OpenReadTransaction();
+    }
+
+    protected override Task WriteItems(DocumentsOperationContext context, AsyncBlittableJsonTextWriter writer)
+    {
+        var revisions = string.IsNullOrEmpty(Collection)
+            ? RequestHandler.Database.DocumentsStorage.RevisionsStorage.GetRevisionsInReverseEtagOrder(context, _start, _pageSize)
+            : RequestHandler.Database.DocumentsStorage.RevisionsStorage.GetRevisionsInReverseEtagOrderForCollection(context, Collection, _start, _pageSize);
+
+        writer.WriteStartArray();
+
+        var first = true;
+        foreach (var revision in revisions)
+        {
+            if (first)
+                first = false;
+            else
+                writer.WriteComma();
+
+            writer.WriteStartObject();
+
+            writer.WritePropertyName(nameof(Document.Id));
+            writer.WriteString(revision.Id);
+            writer.WriteComma();
+
+            writer.WritePropertyName(nameof(Document.Etag));
+            writer.WriteInteger(revision.Etag);
+            writer.WriteComma();
+
+            writer.WritePropertyName(nameof(Document.LastModified));
+            writer.WriteDateTime(revision.LastModified, true);
+            writer.WriteComma();
+
+            writer.WritePropertyName(nameof(Document.ChangeVector));
+            writer.WriteString(revision.ChangeVector);
+            writer.WriteComma();
+
+            writer.WritePropertyName(nameof(Document.Flags));
+            writer.WriteString(revision.Flags.ToString());
+
+            writer.WriteEndObject();
+        }
+
+        writer.WriteEndArray();
```

Comment:
Would be nice we could unify this code with the sharded one...
for that I guess we will need to project the revision to something like
(Id, Etag, LastModified, ChangeVector, Flags)

Reply:
handle rest of the comments first and ping about this one at the end

Reply:
Talked about it and decided to pass